displayer.py

```diff
@@ -1,73 +1,115 @@
-    imgs = [os.path.join(path, obj) for obj in imgs]
-
-    frameind = 0
-    idind = 1
-    x1ind = 2
-    y1ind = 3
-    x2ind = 4
-    y2ind = 5
-    delimiter = ','
-
-    with open(name + ".txt", "r") as fildat:
-        rawdata = fildat.read()
-
-    colors = {1: (255, 0, 0), "ethan": (0, 255, 0), 0: (0, 0, 255)}
-
-    height, width, _ = cv2.imread(imgs[0]).shape
-    out = cv2.VideoWriter(name + '.avi',
-                          cv2.VideoWriter_fourcc('M', 'J', 'P', 'G'),
-                          30 / interval, (width, height))
-
-    rawdata = rawdata.split("\n")
-    index = 0
-    cv2.namedWindow("vid", cv2.WINDOW_NORMAL)
-    cv2.resizeWindow("vid", 1067, 600)
-
-    while index < len(rawdata):
-        line = rawdata[index]
-        ref = line.split(delimiter)
-        next = ref
-        if ref[0] == '':
-            break
-        imgfile = imgs[int(ref[0]) - interval]
-        # print(imgfile)
-        image = cv2.imread(imgfile)
-        while ref[0] == next[0]:
-            # print(next[1], next[2], next[3], next[4])
-            box = ((int(next[x1ind]), int(next[y1ind])), (int(next[x2ind]),
-                                                          int(next[y2ind])))
-            cv2.rectangle(image,
-                          box[0],
-                          box[1],
-                          color=(0, 255, 0),
-                          thickness=3)  # Draw Rectangle with the coordinates
-            cv2.putText(image,
-                        str(next[idind]),
-                        box[0],
-                        cv2.FONT_HERSHEY_SIMPLEX,
-                        3,
-                        color=(0, 255, 0),
-                        thickness=3)
-            index = index + 1
-            next = rawdata[index].split(",")
-
-        out.write(image)
-        cv2.imshow("vid", image)
-        cv2.waitKey(100)
-
-    out.release()
-
-
-if __name__ == "__main__":
-    sys.exit(main())
+import cv2
+import os
+import sys
+import numpy as np
+import argparse
+from tqdm import tqdm
+
+# TODO: (nhendy) this script needs massive clean up
+
+frameind = 0
+idind = 1
+x1ind = 2
+y1ind = 3
+x2ind = 4
+y2ind = 5
+delimiter = ','
+
+def main():
+    args = init_args()
+    create_vid(args.dest_vid, args.source_vid, args.ids_txt, view=args.view)
+
+
+def init_args():
+    parser = argparse.ArgumentParser(description="paints videos with bounding boxes and IDs")
+    parser.add_argument("-v", "--view", default=False, action='store_true')
+    parser.add_argument("source_vid", metavar='src', type=str,
+                        help="the source video to use")
+    parser.add_argument("ids_txt", metavar='ids', type=str,
+                        help="the text file of format <frame, id, x1, y1, x2, y2>")
+    parser.add_argument("dest_vid", metavar="dest", type=str,
+                        help="the name of the destination video")
+    return parser.parse_args()
+
+def create_vid(savename, vid, outtxt, view=False):
```

Comment:
generally let's stick to the convention of using. underscored variable names to delimit words in a variable name.
So savename -> save_name, outtxt -> out_txt ...etc